src/main/host/managed_thread.rs

```diff
@@ -605,6 +603,18 @@ impl ManagedThread {
         let mut file_actions: libc::posix_spawn_file_actions_t = shadow_pod::zeroed();
         Errno::result(unsafe { libc::posix_spawn_file_actions_init(&mut file_actions) }).unwrap();
 
+        // Set up stdin
+        let (stdin_reader, stdin_writer) =
+            nix::unistd::pipe2(nix::fcntl::OFlag::O_CLOEXEC).unwrap();
```

Comment:
maybe create the pipe in `O_DIRECT` mode to guarantee that the entire struct will be read in a single `read` call?

Reply:
From my reading of the man page, I don't think it would make a difference. It seems more useful for a scenario where  we were writing multiple variable-sized objects and wanted to prevent reading e.g. all of the first object and some of subsequent objects. (In particular everything should already be done in one write call and one read call if the data is < PIPE_BUF bytes, and we'll end up failing otherwise)